src/app/devices/deviceContent/components/deviceEvents/deviceEvents.tsx

```diff
@@ -224,98 +211,97 @@ export default class DeviceEventsComponent extends React.Component<DeviceEventsD
-                            ariaLabel={context.t(ResourceKeys.deviceEvents.customEventHub.name.label)}
+                            label={t(ResourceKeys.deviceEvents.customEventHub.name.label)}
+                            ariaLabel={t(ResourceKeys.deviceEvents.customEventHub.name.label)}
                             underlined={true}
-                            value={this.state.customEventHubName}
-                            disabled={this.state.monitoringData}
+                            value={state.customEventHubName}
+                            disabled={state.monitoringData}
                             onChange={customEventHubNameChange}
                             required={true}
                         />
                     </>
                 }
             </>
         );
-    }
+    };
 
-    private stopMonitoring = () => {
-        clearTimeout(this.timerID);
+    const stopMonitoring = async () => {
+        clearTimeout(timerID);
         return stopMonitoringEvents();
-    }
+    };
 
-    private onToggleStart = () => {
-        const monitoringState = this.state.monitoringData;
+    const onToggleStart = () => {
+        const monitoringState = state.monitoringData;
 
         if (monitoringState) {
-            this.stopMonitoring().then(() => {
-                this.setState({
+            stopMonitoring().then(() => {
+                setState({
+                    ...state,
                     monitoringData: false,
                     synchronizationStatus: SynchronizationStatus.fetched
                 });
             });
-            this.setState({
+            setState({
+                ...state,
                 hasMore: false,
                 synchronizationStatus: SynchronizationStatus.updating
             });
         } else {
-            this.setState({
+            setState({
+                ...state,
                 hasMore: true,
                 loading: false,
                 loadingAnnounced: undefined,
                 monitoringData: true
             });
         }
-    }
+    };
 
-    public componentDidMount() {
-        this.isComponentMounted = true;
-    }
-
-    private readonly renderInfiniteScroll = (context: LocalizationContextInterface) => {
-        const { hasMore } = this.state;
+    const renderInfiniteScroll = () => {
+        const { hasMore } = state;
         const InfiniteScroll = require('react-infinite-scroller'); // https://github.com/CassetteRocks/react-infinite-scroller/issues/110
         return (
             <InfiniteScroll
                 key="scroll"
                 className="device-events-container"
                 pageStart={0}
-                loadMore={this.fetchData(context)}
+                loadMore={fetchData()}
                 hasMore={hasMore}
-                loader={this.renderLoader(context)}
-                role={this.state.events && this.state.events.length === 0 ? 'main' : 'feed'}
+                loader={renderLoader()}
```

Comment:
this loader always shows even after user click on 'stop' button. Need to take a closer look on this one.